src/Main.elm

```diff
@@ -49,7 +49,7 @@ type alias Flags =
 
 type alias Model =
     { navKey : Browser.Navigation.Key
-    , url : Url.Url
+    , route : Route.Route
```

Comment:
I know that keeping the route as well as page type seems like a duplicating the information. Anyway storing route is better (more type-safe) than storing url. I also think even NoRedInk stopped using Sum type style `Page` type because it's hard to work with and use product (record with multiple nested models). For now there is this slight duplication of information but swith to route helped me to catch one bug and made some parts of implementation simpler.